src/pyclass.rs

```diff
@@ -30,122 +31,159 @@ pub trait PyClass:
     type Frozen: Frozen;
 }
 
-fn into_raw<T>(vec: Vec<T>) -> *mut c_void {
-    Box::into_raw(vec.into_boxed_slice()) as _
-}
-
 pub(crate) fn create_type_object<T>(py: Python<'_>) -> *mut ffi::PyTypeObject
 where
     T: PyClass,
 {
     match unsafe {
-        create_type_object_impl(
-            py,
-            T::DOC,
-            T::MODULE,
-            T::NAME,
-            T::BaseType::type_object_raw(py),
-            std::mem::size_of::<T::Layout>(),
-            tp_dealloc::<T>,
-            T::dict_offset(),
-            T::weaklist_offset(),
-            T::items_iter,
-            T::IS_BASETYPE,
-            T::IS_MAPPING,
-        )
+        PyTypeBuilder::default()
+            .with_type_doc(T::DOC)
+            .with_offsets(T::dict_offset(), T::weaklist_offset())
+            .with_slot(ffi::Py_tp_base, T::BaseType::type_object_raw(py))
+            .with_slot(ffi::Py_tp_dealloc, tp_dealloc::<T> as *mut c_void)
+            .set_is_basetype(T::IS_BASETYPE)
+            .set_is_mapping(T::IS_MAPPING)
+            .with_class_items(T::items_iter())
```

Comment:
I was under the impression that in Rust usually builder pattern naming conversions are:

* `with_xxx(val: T) -> Self` is an associate method for constructing the builder, for example `fn with_type_doc(doc: &'static str)`
* `xxx(mut self, val: T) -> Self` is a setter, no need to use `set_` prefix.

What do you think?

Reply:
Following the example of `std::process::command`, I think dropping `with_` and `set_` prefixes from all methods would be conformant 👍

Reply:
Removing the `set_` from `set_is_basetype` leads to this clippy warning:
```
warning: methods called `is_*` usually take `self` by mutable reference or `self` by reference or no `self`
   --> src/pyclass.rs:223:19
    |
223 |     fn is_mapping(mut self, is_mapping: bool) -> Self {
    |                   ^^^^^^^^
    |
    = help: consider choosing a less ambiguous name
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention
```

Unless there's an alternative, I'll leave those two as they are, but update the rest.

Reply:
Sounds reasonable to me 👍